pkg/localmetrics/localmetrics.go

```diff
@@ -53,59 +56,57 @@ var (
 )
 
 // UpdateAPIMetrics updates all API endpoint metrics ever 5 minutes
-func UpdateAPIMetrics(APIkey string) {
+func UpdateAPIMetrics(APIKey string, timer *prometheus.Timer) {
 	d := time.Tick(5 * time.Minute)
 	for range d {
-		UpdateMetricPagerDutyHeartbeatGauge(APIkey)
+		UpdateMetricPagerDutyHeartbeat(APIKey, timer)
 	}
 
 }
 
 // UpdateMetricPagerDutyCreateFailure updates guage to 1 when creation fails
 func UpdateMetricPagerDutyCreateFailure(x int, cd string) {
 	MetricPagerDutyCreateFailure.With(prometheus.Labels{
-		"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(
+		"clusterdeployment_name": cd}).Set(
 		float64(x))
 }
 
 // UpdateMetricPagerDutyDeleteFailure updates guage to 1 when deletion fails
 func UpdateMetricPagerDutyDeleteFailure(x int, cd string) {
 	MetricPagerDutyDeleteFailure.With(prometheus.Labels{
-		"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(
+		"clusterdeployment_name": cd}).Set(
 		float64(x))
 }
 
-// UpdateMetricPagerDutyHeartbeatGauge curls the PD API, updates the gauge to 1
+// UpdateMetricPagerDutyHeartbeat curls the PD API, updates the gauge to 1
 // when successful.
-func UpdateMetricPagerDutyHeartbeatGauge(APIkey string) {
+func UpdateMetricPagerDutyHeartbeat(APIKey string, timer *prometheus.Timer) {
 	metricLogger := log.WithValues("Namespace", "pagerduty-operator")
 	metricLogger.Info("Metrics for PD API")
 
 	_, err := http.NewRequest("GET", "api.pagerduty.com", nil)
 	if err != nil {
-		MetricPagerDutyHeartbeat.With(prometheus.Labels{
-			"name": "pagerduty-operator"}).Set(float64(0))
+		MetricPagerDutyHeartbeat.Observe(float64(timer.ObserveDuration()))
 		metricLogger.Error(err, "Failed to get reach api")
 	}
 
-	// if there is an api key make an authenticated calld
-	if APIkey != "" {
+	// if there is an api key make an authenticated called
+	if APIKey != "" {
 		req, _ := http.NewRequest("GET", apiEndpoint, nil)
 		req.Header.Set("Accept", "application/vnd.pagerduty+json;version=2")
 		req.Header.Set("Content-Type", "application/json")
-		req.Header.Set("Authorization", "Token token="+APIkey)
-
+		req.Header.Set("Authorization", "Token token="+APIKey)
```

Comment:
I prefer `fmt.Sprintf("Token token=%s", APIKey)` here as it feels more explicit and easier to read.

Reply:
I'll take that suggestion